src/connectedk8s/HISTORY.rst

```diff
@@ -3,13 +3,16 @@
 Release History
 ===============
 
+1.5.0
+++++++
+* Enforce valid custom locations service principal application object id passed in by the user for enabling custom locations feature.
```

Comment:
I would like to confirm that this change will not cause a breaking change for the customer, right?

Reply:
This is breaking for Azure Cloud users who expect the code to pick the right SP application to use for custom locations feature even though custom locations SP app object id (`--custom-locations-oid`) was not passed, which is the wrong behavior to begin with. This update is a breaking change to enforce correct behavior.

Here is the old code:
https://github.com/Azure/azure-cli-extensions/blob/419c252ee7ff0d85203ef3b6cf19ca71ac30488a/src/connectedk8s/azext_connectedk8s/custom.py#L2174C23-L2174C23

The behavior in the old code checks to see if "Custom Locations RP" SP application exists in AAD graph. It should always exist for Azure Cloud, from my understanding. Therefore it goes inside the `if len(result) != 0:` block. Inside that block, it checks `if cl_oid is not None and cl_oid != result[0].object_id:`. If user didn't pass `cl_oid` (parameter `--custom-locations-oid`), then `return result[0].object_id` is returned anyway even if user has no intention to enable custom locations. This ends up enabling custom locations and is the wrong behaviour since user didn't pass `--custom-locations-oid`. Therefore, currently, custom locations feature is always enabled as long as customer has an SP app named "Custom Locations RP", and this is always the case for Azure Cloud (to my understanding), which is the wrong behaviour.

Reply:
Have you evaluated the impact of this breaking change on CLI customers?

Reply:
This will impact Azure Arc users. But there is no loss of functionality, they just need to explicitly pass "--custom-locations-oid" to enable custom locations. Otherwise, it is not enabled by default. Previous code guesses the custom locations object ID via display name of an SP app, which is not unique and may result in buggy behavior.

Reply:
Yes, I understand. Could we first add the warning logs to guide customers to migrate their usage, and implement the breaking change in the breaking change window of CLI (**Ignite Sprint**)? I believe this will minimize the impact on customers, what do you think of it?

Reply:
Hi @zhoxing-ms, this warning already exists:
https://github.com/TheOnlyWei/azure-cli-extensions/blob/b315d43fe8c51d47b8cd2f05748e5bed1eec9852/src/connectedk8s/azext_connectedk8s/custom.py#L2154C31-L2154C31

Reply:
Got it, thanks~